configure

```diff
@@ -3475,6 +3475,7 @@ ogg_demuxer_select="dirac_parse"
 ogv_muxer_select="ogg_muxer"
 opus_muxer_select="ogg_muxer"
 psp_muxer_select="mov_muxer"
+rtc_muxer_deps_any="openssl"
```

Comment:
Kieran Kunhya:

This patchset is massive and unmaintainable. Not to mention only implements a small portion of webrtc, that will inevitably get patched to add more features and become a giant sprawl.

Reply:
If we aim to support all the full features of WebRTC, the patch will become a sprawling giant and difficult to maintain. However, I believe that WHIP is only a small subset of WebRTC, as there are some usefull features that we do not need to support. For example:

* P2P and ICE will require the inclusion of numerous transport algorithms and libraries. However, we only need to support ICE Lite, which functions as a UDP client and server, similar to other protocols like SRT, RTMP, or RTSP that are supported by FFmpeg.
* Simulcast and SVC are also significant gitant features, and I do not recommend supporting them immediately. These advanced features are not necessary, and they are not commonly used in WebRTC scenarios.
* Congestion control and algorithms, such as ARQ and FEC, can be straightforward if we only support NACK for ARQ and simple FEC like RED. However, supporting GCC or RSFEC, etc., can be complex. I also suggest not supporting them immediately because, in some use scenarios, the network conditions are good.
* Signaling and room management are not necessary to touch as WHIP is a straightforward signaling method for WebRTC. FFmpeg only needs to publish a WebRTC stream for use cases such as robot control or similar scenarios. Therefore, there is no need to introduce rooms or multiple streams in a WebRTC session.
* Audio processing, such as 3A, is not required for FFmpeg as it is a feature for players like Chrome.

This patch is simple not only because WHIP is a subset of WebRTC, but also because it reuses many of FFmpeg's powerful capabilities, such as codec and format, SRTP, HTTP, HTTPS, OpenSSL, and more.

如果我们要支持完善的WebRTC的功能，那么这个会变得很大而且难以维护。但我理解WHIP只是WebRTC的一个子集，如果我们只是实现有限但有用的功能，那么最终是可维护的，比如：

* P2P和ICE Full是复杂的功能，但是WHIP只需要实现ICE lite。这样在网络层只需要实现类似CS结构，和SRT、RTMP、RTSP等目前已经实现的协议是类似的。
* Simulcast和SVC这是WebRTC中比较复杂的高阶能力，一般涉及多流和网络自适应调整，但它并不是必须的功能，非常非常多的应用场景并不需要这些高阶能力。
* Congestion  control拥塞控制算法，比如ARQ和FEC。如果只是实现基本的NACK和FEC复杂度是可控的。如果要实现高阶的GCC和RSFEC等会很复杂。我建议先不要着急实现复杂的算法，一般网络条件很好，也并不需要完全实现这些复杂的算法。
* Signaling和Room管理，WHIP是非常简单的信令，我们并不需要实现WebRTC业务中常见的复杂信令和房间管理。FFmpeg是基于流级别的处理，房间中的多个流本来也可以解耦成独立的流。
* Audio处理，比如3A，这些是播放器需要处理的功能，FFmpeg本身也不需要实现。

这个Patch看起来简单，不仅仅是因为WHIP是WebRTC的一个子集，更加重要的是复用了FFmpeg强大的现有的能力，比如编解码和封装、SRTP、HTTP、HTTPS和OpenSSL等。